### PR TITLE
Add swp token distribution

### PR DESCRIPTION
**Pull request by karzak**
Adds SWP token distribution according to token supply schedule specified in prop 59. 

Tests verify total supply of SWP token and the expected spendable balance of important accounts at genesis time.

### CODE REVIEW

File: migrate/v0_15/migrate.go [hunk 1]
```diff
@@ -141,6 +148,96 @@ func MigrateAccounts(genesisState auth.GenesisState, genesisTime time.Time) auth
...
 
+// DistributeSwpTokens sets the initial distribution of swp tokens according to the proposed token supply schedule in prop 59
+func DistributeSwpTokens(genesisState auth.GenesisState) auth.GenesisState {
+	accounts := make([]authexported.GenesisAccount, len(genesisState.Accounts))
+	migratedGenesisState := auth.NewGenesisState(genesisState.Params, accounts)
+	// add SWP incentives (LP, Kava stakers) to kavadist module account
+	kavaDistAddr := supply.NewModuleAddress(kavadist.KavaDistMacc)
+	for i, acc := range genesisState.Accounts {
+		if acc.GetAddress().Equals(kavaDistAddr) {
+			err := acc.SetCoins(acc.GetCoins().Add(sdk.NewCoin("swp", sdk.NewInt(137500000e6))))
```
Comment (denalimarsh) [inline migrate/v0_15/migrate.go:159]:
```suggestion
			err := acc.SetCoins(acc.GetCoins().Add(sdk.NewCoin("swp", sdk.NewInt(kavaDistAddrSwpAmount))))
```

nit: making the SWP token amounts constants defined at the top of the file or function would be clearer.

File: migrate/v0_15/migrate.go [hunk 2]
```diff
@@ -18,13 +18,14 @@ import (
...
 	v0_14validator_vesting "github.com/kava-labs/kava/x/validator-vesting"
 )
 
 var (
 	// TODO: update GenesisTime and chain-id for kava-8 launch
-	GenesisTime = time.Date(2021, 8, 30, 15, 0, 0, 0, time.UTC)
+	GenesisTime = time.Date(2021, 8, 30, 15, 0, 0, 0, time.UTC) // 1630335600
 	ChainID     = "kava-8"
 	// TODO: update SWP reward per second amount before production
 	// TODO: add swap tokens to kavadist module account
```
Comment (denalimarsh) [inline migrate/v0_15/migrate.go:31]: Can remove this TODO now

File: migrate/v0_15/migrate.go [hunk 3]
```diff
@@ -141,6 +148,96 @@ func MigrateAccounts(genesisState auth.GenesisState, genesisTime time.Time) auth
...
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+		sdk.NewCoin("swp", sdk.NewInt(5859375e6)),
+	}
+	swpTreasuryVestingLengths := []int64{7948800, 7776000, 7948800, 7862400, 7948800, 7776000, 7948800, 7862400}
```
Comment (denalimarsh) [inline migrate/v0_15/migrate.go:229]: Why is there variation in treasury vesting lengths when the token amount for each is the same?

File: migrate/v0_15/migrate.go (hunk 1 above)
Comment (denalimarsh) [reply migrate/v0_15/migrate.go:159]: Could also then use the values to test individual addresses' token balances

**Review by denalimarsh — COMMENTED** [review]
Changes generally look good. Just a few nits/questions/recommendations.

File: migrate/v0_15/migrate.go (hunk 3 above)
Comment (karzak) [reply migrate/v0_15/migrate.go:229]: Vesting dates fall at the end of the month. So for months like Feb, vesting lengths are a bit shorter, etc

**Review by rhuairahrighairidh — APPROVED** [review]
looks good, couldn't find any problems

areas checked over:
- new coin amounts and vesting periods against proposal
- walk through of `DistributeSwpTokens`
- check of slice/pointer manipulation in `MigrateAccounts`, `ApplySwpAirdrop`, `DistributeSwpTokens`
- tested original coins balances were not modified by `Auth`

I put the test I wrote in #1005.
It also contains a small refactor in `DistributeSwpTokens` to make the slice manipulation a bit safer.

**Pull request merged**